processors/tracecompletiontime-deriver/src/main/java/org/hawkular/apm/processor/tracecompletiontime/TraceCompletionInformationInitiator.java

```diff
@@ -78,7 +77,7 @@ public TraceCompletionInformation processOneToOne(String tenantId,
                 ci.setCompletionTime(ct);
 
                 // Initialise any communications that need to be further processed
-                TraceCompletionInformationUtil.initialiseCommunications(ci, item.getNodes().get(0).getBaseTime(),
+                TraceCompletionInformationUtil.initialiseLinks(ci, item.getNodes().get(0).getBaseTime(),
```

Comment:
Well spotted  - I think I originally thought the same method would be used with subsequent services in the chain, so the baseDuration would be incrementing.